src/components/IconPicker.tsx

```diff
@@ -52,15 +52,21 @@ const IconPicker = ({ schemaType, value = {}, onChange }: ObjectInputProps) => {
     setSelected(null);
   };
 
-  const setIcon: SearchResultsOnSelectCallback = (icon) => {
+  const setIcon: SearchResultsOnSelectCallback = (
+    icon: IconObject,
+    ele: HTMLButtonElement
+  ) => {
     if (selected && icon.name === selected.name) return unsetIcon();
 
+    const svgString = ele.getElementsByTagName('svg')[0].outerHTML;
+
     onChange([
       setIfMissing({
         _type: schemaType.name,
       }),
       set(icon.name, ['name']),
       set(icon.provider, ['provider']),
+      schemaType.options.storeSvg ? set(svgString, ['svg']) : unset(['svg']),
```

Comment:
And based on the change above

```suggestion
      schemaType.options.storeSvg ? set(svgString(), ['svg']) : unset(['svg']),
```